reverse_file_buffered.py

```diff
@@ -0,0 +1,36 @@
+from file_io import Io, FileIo
+
+
+def reverse_file_in_memory(io: Io, name: str, buf_size: int):
+    """Reverse a content of the file `name` using no more than buf_size of additional memory"""
+    assert buf_size >= 2, "Minimum allowed buf_size is 2"
+
+    length = io.file_length(name)
+    byte_size = buf_size//2
+    output_buff = bytearray(buf_size)
+    left_offset = 0
+    right_offset = length - byte_size
```

Comment:
Is this always correct?

Reply:
1. in case buf_size is an odd number we don't use all allocated memory while calculating chunk_size like buf_size//2 (extra byte remaining)
2. there could also be some issues with offsets logic as I see some overlaps in characters reversing.